File: KPCJumpBarControlDemo/ViewController.m
//
//  ViewController.m
//  KPCJumpBarControlDemo
//
//  Created by Cédric Foellmi on 01/05/16.
//  Copyright © 2016 onekiloparsec. All rights reserved.
//

#import "ViewController.h"
#import <KPCJumpBarControl/KPCJumpBarControl.h>

@interface ViewController () <KPCJumpBarControlDelegate>
@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.selectedItemIcon.image = nil;
    self.selectedItemTitle.stringValue = @"";
    self.selectedItemIndexPath.stringValue = @"";

    KPCJumpBarItem *rootSegment = [KPCJumpBarItem itemWithTitle:@"level 0" icon:[NSImage imageNamed:@"Oval"]];
    
    KPCJumpBarItem *segment1Item0 = [KPCJumpBarItem itemWithTitle:@"level 1.0" icon:[NSImage imageNamed:@"Polygon"]];
    KPCJumpBarItem *segment1Item1 = [KPCJumpBarItem itemWithTitle:@"level 1.1" icon:[NSImage imageNamed:@"Rectangle"]];
    KPCJumpBarItem *segment1Item2 = [KPCJumpBarItem itemWithTitle:@"level 1.2" icon:[NSImage imageNamed:@"Triangle"]];
    
    rootSegment.children = @[segment1Item0, segment1Item1, segment1Item2];

    KPCJumpBarItem *segment2Item0 = [KPCJumpBarItem itemWithTitle:@"level 2.0" icon:[NSImage imageNamed:@"Star"]];
    KPCJumpBarItem *segment2Item1 = [KPCJumpBarItem itemWithTitle:@"level 2.1" icon:[NSImage imageNamed:@"Spiral"]];
    
    segment1Item1.children = @[segment2Item0, segment2Item1];

    [self.jumpBar useItemsTree:@[rootSegment]];
    
    self.jumpBar.delegate = self;
}

#pragma mark - KPCJumpBarControlDelegate 

- (void)jumpBarControl:(KPCJumpBarControl * _Nonnull)jumpBar willOpenMenuAtIndexPath:(NSIndexPath * _Nonnull)indexPath withItems:(NSArray <id<KPCJumpBarItem>> * _Nonnull)items
{
    NSLog(@"%@", NSStringFromSelector(_cmd));
}

- (void)jumpBarControl:(KPCJumpBarControl * _Nonnull)jumpBar didOpenMenuAtIndexPath:(NSIndexPath * _Nonnull)indexPath withItems:(NSArray <id<KPCJumpBarItem>> * _Nonnull)items
{
    NSLog(@"%@", NSStringFromSelector(_cmd));
}

- (void)jumpBarControl:(KPCJumpBarControl * _Nonnull)jumpBar willSelectItem:(id<KPCJumpBarItem> _Nonnull)item atIndexPath:(NSIndexPath * _Nonnull)indexPath
{
    NSLog(@"%@", NSStringFromSelector(_cmd));
}

- (void)jumpBarControl:(KPCJumpBarControl * _Nonnull)jumpBar didSelectItem:(id<KPCJumpBarItem> _Nonnull)item atIndexPath:(NSIndexPath * _Nonnull)indexPath
{
    NSLog(@"%@", NSStringFromSelector(_cmd));
    
    self.selectedItemIcon.image = [item icon];
    self.selectedItemTitle.stringValue = [item title];
    self.selectedItemIndexPath.stringValue = [@"IndexPath: " stringByAppendingString:[indexPath KPC_stringValue]];
}


@end

File: KPCJumpBarControlDemo/ViewController.swift
//
//  ViewController.swift
//  KPCJumpBarControlDemo
//
//  Created by Cédric Foellmi on 09/05/16.
//  Copyright © 2016 onekiloparsec. All rights reserved.
//

import Cocoa
import KPCJumpBarControl

class ViewController: NSViewController, JumpBarControlDelegate {

    @IBOutlet weak var jumpBar: JumpBarControl? = nil
    @IBOutlet weak var selectedItemTitle: NSTextField? = nil
    @IBOutlet weak var selectedItemIcon: NSImageView? = nil
    @IBOutlet weak var selectedItemIndexPath: NSTextField? = nil
    
    var swap: Bool = true

    override func viewDidLoad() {
        super.viewDidLoad()

        self.selectedItemIcon?.image = nil
        self.selectedItemTitle?.stringValue = ""
        self.selectedItemIndexPath?.stringValue = ""
    
        self.jumpBar?.delegate = self
        self.swapItemsTree()
    }
    
    func swapItemsTree() {
        if swap == true {
            let rootSegment = JumpBarItem(withTitle:"path 0", icon:NSImage(named:NSImage.Name("Oval")))
            
            let segment1Item0 = JumpBarItem(withTitle:"path 0.0", icon:NSImage(named:NSImage.Name("Polygon")))
            let segment1Item1 = JumpBarItem(withTitle:"path 0.1", icon:NSImage(named:NSImage.Name("Rectangle")))
            let separatorItem = JumpBarItem.separatorItem()
            let segment1Item2 = JumpBarItem(withTitle:"path 0.3 - switch to another tree", icon:NSImage(named:NSImage.Name("Triangle")))

            rootSegment.children = [segment1Item0, segment1Item1, separatorItem, segment1Item2]

            let segment2Item0 = JumpBarItem(withTitle:"path 0.1.0", icon:NSImage(named:NSImage.Name("Star")))
            let segment2Item1 = JumpBarItem(withTitle:"path 0.1.1", icon:NSImage(named:NSImage.Name("Spiral")))

            segment1Item1.children = [segment2Item0, segment2Item1]
            
            self.jumpBar?.useItemsTree([rootSegment])
        }
        else {
            let rootSegment0 = JumpBarItem(withTitle:"path 0", icon:NSImage(named:NSImage.Name("Rectangle")))
            let rootSegment1 = JumpBarItem(withTitle:"path 1", icon:NSImage(named:NSImage.Name("Star")))

            let segment1Item0 = JumpBarItem(withTitle:"path 1.0", icon:NSImage(named:NSImage.Name("Polygon")))
            rootSegment1.children = [segment1Item0]

            let segment2Item0 = JumpBarItem(withTitle:"path 1.0.0", icon:NSImage(named:NSImage.Name("Spiral")))
            segment1Item0.children = [segment2Item0]

            let segment3Item0 = JumpBarItem(withTitle:"path 1.0.0.0 - switch to another tree", icon:NSImage(named:NSImage.Name("Triangle")))
            segment2Item0.children = [segment3Item0]

            self.jumpBar?.useItemsTree([rootSegment0, rootSegment1])
        }
    }

    // MARK: - JumpBarControlDelegate

     func jumpBarControl(_ jumpBar: JumpBarControl, willOpenMenuAtIndexPath indexPath: IndexPath, withItems items: [JumpBarItemProtocol]) {
        print(#function)
    }

     func jumpBarControl(_ jumpBar: JumpBarControl, didOpenMenuAtIndexPath indexPath: IndexPath, withItems items: [JumpBarItemProtocol]) {
        print(#function)
    }

     func jumpBarControl(_ jumpBar: JumpBarControl, willSelectItems items: [JumpBarItemProtocol], atIndexPaths indexPaths: [IndexPath]) {
        print(#function)
    }

     func jumpBarControl(_ jumpBar: JumpBarControl, didSelectItems items: [JumpBarItemProtocol], atIndexPaths indexPaths: [IndexPath]) {
        print(#function)
        
        self.selectedItemIcon?.image = (items.count == 1) ? items.first!.icon : nil
        self.selectedItemTitle?.stringValue = (items.count == 1) ? items.first!.title : "(multiple selection)"
        self.selectedItemIndexPath?.stringValue = (items.count == 1) ? "IndexPath: \(indexPaths.first!.description)" : "(multiple selection)"

        if indexPaths.count == 1 && (indexPaths.first! == IndexPath(indexes: [0, 3]) || indexPaths.first! == IndexPath(indexes: [1, 0, 0, 0])) {
            swap = !swap
            self.swapItemsTree()
        }
    }
}

